src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -2205,7 +2205,7 @@ public String getHelpFile() {
          * @param parameters the parameters for the new build, used to find it later.
          * @param projectName the name of the current project for better logging.
          */
-        public synchronized void scheduled(ChangeBasedEvent event, ParametersAction parameters, String projectName) {
+        public void scheduled(ChangeBasedEvent event, ParametersAction parameters, String projectName) {
```

Comment:
Removing the `synchronized` means risk of `ConcurrentModificationException` below when iterating over `runningJobs`

Reply:
Well maybe not `ConcurrentModificationException` but there is no guard against the map being manipulated while iterating it iiuc.

Reply:
I will take a look, this code is in our production for 3 weeks already and everything was fine (at least we did not face deadlocks or CME). But I agree that some guarding can be missed.

Reply:
You were right, we observed non-deterministic behavior during big load (then we start to receive tons of events from here and there). I submitted a new patch, but (!) I think it would be better to wait at least 1 week before merging it :)

Reply:
@rsandell So far looks good.